metals/src/main/scala/scala/meta/internal/metals/BloopInstall.scala

```diff
@@ -243,9 +244,14 @@ object BloopInstall {
         fn: String => Unit
     ): Unit = {
       if (!closed) {
-        val text = toPlainString(buffer).trim
-        if (text.nonEmpty) {
-          fn(text)
+        val text = toPlainString(buffer)
+        output = output + text
+        while (output.contains('\n')) {
```

Comment:
`output.indexOf('\n') > 0` ? no need to follow up `contains` with the `indexOf`. Asa a result we can also drop the inner `if` :)

Reply:
or maybe something like:
```
val lines = output.splitBy(char == '\n') // if no '\n' then lines == Array(output)
lines.dropRight(1).forEach(fn)
output = lines.last  // last entry is everything after the last '\n'
```

Reply:
Is it possible to use `String.linesIterator.foreach(...)` in this situation?

Reply:
the problems is that `"asbcd\n".linesIterator.size == 1` and I need to differentiate between "asbcd\n" and "asbcd" situation. I think this while is a more straightforward solution

Reply:
also if I do it this way I will always have everything written properly till the end since build tools always end with \n

Reply:
Here's an implementation that doesn't use while, vars or string concat

```scala
@   def forallLines(text: String, fn: String => Unit): Unit = {
      def loop(start: Int): Unit = {
        val nl = text.indexOf('\n', start)
        if (nl < 0) {
          fn(text.substring(start))
        } else {
          fn(text.substring(start, nl))
          loop(nl + 1) // hop over newline character
        }
      }
      loop(0)
    }
defined function forallLines

@ forallLines("a\nb\n", line => pprint.log(line))
ammonite.$sess.cmd9.res9:1 line: "a"
ammonite.$sess.cmd9.res9:1 line: "b"
ammonite.$sess.cmd9.res9:1 line: ""
```

Reply:
I'm concerned this doesn't handle `\r\n` however, which `String.linesIterator` does. I would recommend using `linesIterator` instead and if the trailing endline is important then you can add a special case for it
```scala
output.linesIterator.foreach(fn)
if (output.endsWith("\n")) fn("")
```

Reply:
That var is a buffer between invocations though, that's why it's important. Maven send a single line in 3 parts. I did switch it to StringBuilder which should be better for now. As for lineIterator I will check what will happen with `\r\n`.

Reply:
Is it necessary to do `fn(line)` on each individual line? What happens if we call `scribe.info` on messages with multiple newlines?

Reply:
Ok, added using lineIter and I also check if we should include the last line depending if it ends with \n